apps/avifgainmaputil/swapbase_command.cc

```diff
@@ -18,20 +18,13 @@ avifResult ChangeBase(avifImage& image, avifImage* swapped) {
       image.gainMap->metadata.alternateHdrHeadroomD;
   const bool tone_mapping_to_sdr = (headroom == 0.0f);
 
-  // The gain map's cicp values are those of the 'tmap' item and describe the
-  // image obtained by fully applying the gain map. See ISO/IEC JTC 1/SC 29/WG 3
-  // m64379v1 4.1.1: A tmap derived item shall be associated with a colr item
-  // property. This property describes the colour properties of the
-  // reconstructed image if the gain map input item is fully applied according
-  // to ISO/AWI 214961.
-  if (image.gainMap->image->transferCharacteristics !=
-          AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED ||
-      image.gainMap->image->colorPrimaries !=
-          AVIF_COLOR_PRIMARIES_UNSPECIFIED) {
-    swapped->colorPrimaries = image.gainMap->image->colorPrimaries;
+  if (image.gainMap->altColorPrimaries != AVIF_COLOR_PRIMARIES_UNSPECIFIED ||
+      image.gainMap->altTransferCharacteristics !=
+          AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED) {
```

Comment:
`matrixCoefficients` look like they were forgotten from the condition but I guess not, maybe add a comment why?

Reply:
Yeah it was intentional but a bit weird indeed. Reworked.